client/state/selectors/can-current-user-manage-plugins.js

```diff
@@ -0,0 +1,22 @@
+/**
+ * Internal dependencies
+ */
+import createSelector from 'lib/create-selector';
+import { canCurrentUser } from 'state/selectors';
+
+/**
+ * Returns true if user can manage plugins for at least one site and returns false otherwise
+ *
+ * @param {Object} state  Global state tree
+ * @return {Boolean} Whether the user can manage plugins or not
+ */
+export default createSelector(
+	state => {
+		if ( ! state || ! state.currentUser || ! state.currentUser.capabilities ) {
+			return false;
+		}
+		const siteIds = Object.keys( state.currentUser.capabilities );
+		return siteIds.some( ( siteId ) => canCurrentUser( state, siteId, 'manage_options' ) );
+	},
+	state => state.currentUser.capabilities
+);
```

Comment:
A couple of notes:

- There is inconsistency in that the selector checks for `! state || ! state.currentUser …` but the function for dependent state (second argument of `createSelector`) doesn't, instead accessing `state.currentUser.capabilities` directly. Also, imagine if all selectors had to check for `! state`. 😅
- In reality, if these property accesses were to fail, that would be a code smell, since we expect our reducers to properly lay our application's initial state. Here, we expect `state ∋ currentUser ∋ capabilities ≔ {}`, [as defined](https://github.com/Automattic/wp-calypso/blob/a3930021c57a0b3199fa60bbeb450fd81db4e7eb/client/state/current-user/reducer.js#L64).
- That said, it's idiomatic to prefer `lodash#get` and provide a type-safe (and semantic) fallback value, e.g.

```js
// `get` will return `{}` in case of failure,
// meaning `siteId` will default to `[]`,
// thus `siteIds.some( f )` will default to `false`, as expected
const siteIds = Object.keys( get( state, 'currentUser.capabilities', {} ) );